ex03.exs

```diff
@@ -59,8 +59,32 @@ defmodule Ex03 do
 
   """
 
-  def pmap(collection, process_count, function) do
-    « your code here »
+  # This exists so that we are always guarenteed a chunk size of at least one
+  # Ex:  [1], n = 100
+  #   div 1, 100 -> 0
+  #   chunk_every with 0 fails
+  defp check_zero(0), do: 1
+  defp check_zero(x), do: x
+
+  defp chunk_collection(collection, n) do
+    chunk_size = div(Enum.count(collection),n) |> check_zero
+    Enum.chunk_every(collection, chunk_size)
+  end
+
+  defp start_task(collection, func) do
+    Task.async(fn -> Enum.map(collection, func) end)
+  end
+
+  # Does this make sense to still run when pcount > collection size?
+  # I see an argument here on both sides - if it's a user facing application
+  # pmap should still work with any size array and any processor number
+  # if it's a developer tool - i think it's fair to assume developers wont pmap
+  # a single element array with 1000 processors.
```

Comment:
## A good question. 

If you assume the collection is a list, you can always add a simple guard clause:

~~~ elixir
def pmap(collection, process_count, func) do when process_count < length(collection) do
~~~

That way the function wouldn't be defined for these invalid cases.